src/main/blackbox/blackbox.c

```diff
@@ -701,6 +708,13 @@ static void writeIntraframe(void)
         }
     }
 
+    if (isFieldEnabled(FIELD_SELECT(SERVO))) {
+        const int servoCount = MAX_SUPPORTED_SERVOS;
```

Comment:
this wrongly assumes that MAX_SUPPORTED_SERVOS is always 8 (see USE_QUAD_MIXER_ONLY). Blackbox content must match BB header exactly.